code/sorting/sort_test/sort_test.cpp

```diff
@@ -2,109 +2,163 @@
+//    vsf = selectionSort;
+//    psf = selectionSort;
+//    vsf = quickSort;
+//    psf = quickSort;
+
     SECTION("test empty") {
         const int sz = 0;
         int *arr = new int[sz];
         int *arr_end = arr+sz;
-        vector<int> vec(sz);
+        ContainerType<int> stdContainer(sz);
         
         psf(arr, arr_end);
-        vsf(vec.begin(), vec.end());
+        vsf(stdContainer.begin(), stdContainer.end());
     }
     
     SECTION("test has 1 elem") {
         const int sz = 1;
-        int *arr = new int[sz];
+        int *arr = new int[sz]{1};
         int *arr_end = arr+sz;
-        vector<int> vec(sz);
-        arr[0] = 1;
-        vec[0] = 1;
+        ContainerType<int> stdContainer{1};
         
         psf(arr, arr_end);
-        vsf(vec.begin(), vec.end());
+        vsf(stdContainer.begin(), stdContainer.end());
         
         CHECK(arr[0] == 1);
-        CHECK(vec[0] == 1);
+        CHECK(*stdContainer.begin() == 1);
     }
     
     SECTION("test has 2 elems") {
         const int sz = 2;
         int *arr = new int[sz]{3, 1};
         int *arr_end = arr+sz;
-        vector<int> vec(sz);
-        vec[0] = 3;
-        vec[1] = 1;
+        ContainerType<int> stdContainer{3, 1};
         
         psf(arr, arr_end);
-        vsf(vec.begin(), vec.end());
+        vsf(stdContainer.begin(), stdContainer.end());
         
         CHECK(arr[0] == 1);
         CHECK(arr[1] == 3);
-        CHECK(vec[0] == 1);
-        CHECK(vec[1] == 3);
+        CHECK(*stdContainer.begin() == 1);
+        CHECK(*++stdContainer.begin() == 3);
+    }
+
+    SECTION("test has 3 elems") {
+        const int sz = 3;
+        int *arr = new int[sz]{2, 3, 1};
+        int *arr_end = arr+sz;
+        ContainerType<int> stdContainer{2, 3, 1};
+
+        psf(arr, arr_end);
+        vsf(stdContainer.begin(), stdContainer.end());
+
+        CHECK(arr[0] == 1);
+        CHECK(arr[1] == 2);
+        CHECK(arr[2] == 3);
+        CHECK(*stdContainer.begin() == 1);
+        CHECK(*++stdContainer.begin() == 2);
+        CHECK(*++++stdContainer.begin() == 3);
     }
     
     SECTION("test has random size elems and random value") {
         srand((int)clock());
         for (int t = 0; t < 5000; ++t) {
-            // random size
+            // randomize size
             int rand = 75 + std::rand()%50;
-            int *actualArr = new int[rand], *expectArr = new int[rand];
-            vector<int> actualVec(rand), expectVec(rand);
 
-            // random elems
-            auto actualIt = actualVec.begin(), expectIt = expectVec.begin();
+            int *actualDynamicArray = new int[rand];
+            ContainerType<int> actualStdContainer{}, expectStdContainer{};
+
+            // randomize elems
+            std::back_insert_iterator<ContainerType<int>> actualIt(actualStdContainer), expectIt(expectStdContainer);
             for (int i = 0; i < rand; ++i, ++actualIt, ++expectIt) {
                 int rv = std::rand()%100;
```

Comment:
Spaces around the operators. Also, this form of random is not recommended.

Reply:
"Also, this form of random is not recommended."
I don't understood your mean.
did you say the variable `rand` name is not recommend?

Reply:
No, that, and C style random number generation is pretty slow, although it shouldn't matter here.